python/tvm/_ffi/ndarray.py

```diff
@@ -343,3 +347,40 @@ def _tvm_handle(self):
         raise ValueError("Cannot register create when extension tcode is same as buildin")
     _reg_extension(cls, fcreate)
     return cls
+
+
+def register_ndarray(cls, fcreate):
```

Comment:
can we simply use register_extension to register extension NDArray?

Reply:
`register_extension` requires `_tvm_code`, while NDArray subclasses requires `_array_type_index`. They are pretty much different from my side. Any suggestions?

Reply:
can we simply detect whether if it is subclass of NDArray and dispatch to different logic?

Reply:
okie, will do